src/http_server.rs

```diff
@@ -0,0 +1,150 @@
+// Copyright 2018 the Deno authors. All rights reserved. MIT license.
+use errors::DenoError;
+use errors::DenoResult;
+
+use futures::sync::mpsc;
+use futures::sync::oneshot;
+use futures::Sink;
+use hyper::rt::{Future, Stream};
+use hyper::service::service_fn;
+use hyper::{Body, Request, Response, Server};
+use std::net::SocketAddr;
+use tokio;
+
+pub type Res = Response<Body>;
+
+// server -> loop
+pub struct Transaction {
+  pub req: Request<Body>,
+  pub response_tx: Option<oneshot::Sender<Res>>,
+}
+
+// accept -> loop
+pub type TransactionSender = oneshot::Sender<Transaction>;
+
+pub struct HttpServer {
+  sender_a: mpsc::Sender<TransactionSender>,
+  complete_tx: Option<oneshot::Sender<()>>,
+}
+
+/// This ensures that when the HttpServer is dropped from the resources table
+/// it will stop its pending future.
+impl Drop for HttpServer {
+  fn drop(&mut self) {
+    let complete_tx = self.complete_tx.take().unwrap();
+    complete_tx.send(()).unwrap();
+  }
+}
+
+impl HttpServer {
+  pub fn accept(&self) -> impl Future<Item = Transaction, Error = DenoError> {
+    let (transaction_sender, transaction_reciever) =
```

Comment:
nit: `reciever` -> `receiver` (and many other places)

Reply:
Fixed